Code quality improvements

lontivero opened the pull request:
This PR contains several code quality improvements detected by the static code analysis tool.

nopara73 commented on WalletWasabi/Helpers/HttpMessageHelper.cs:
```diff
@@ -454,7 +441,7 @@ public static void ParseFistChunkLine(string firstChunkLine, out long chunkSize,
 
 			var allData = new byte[(int)length];
-			var num = await stream.ReadBlockAsync(allData, (int)length);
+			var num = await stream.ReadBlockAsync(allData, (int)length, ctsToken);
```
Not sure about this.

lontivero replied:
Ok, then we could remove the ctsToken from the parameters list.

nopara73 replied:
Yes, please remove it.

nopara73 commented on WalletWasabi/Helpers/HttpMessageHelper.cs:
```diff
@@ -483,7 +470,7 @@ public static void AssertValidHeaders(HttpHeaders messageHeaders, HttpContentHea
+			if (contentHeaders != null && contentHeaders.Contains("Content-Length"))
 			{
 				if (contentHeaders.ContentLength < 0)
 					throw new HttpRequestException("Content-Length MUST be bigger than zero.");
```
Could you change it to "larger than" instead of "bigger than"

lontivero replied:
Sure.

nopara73 commented on WalletWasabi/Http/Models/StartLine.cs:
```diff
@@ -45,11 +45,6 @@ public static List<string> GetParts(string startLineString)
-							throw new Exception($"Wrong {startLineString} provided.");
-						}
-
 						var rest = reader.ReadToEnd();
```
Is this because it can never happen?

nopara73 replied:
Ok, yes, I can see from commit comment.

nopara73 commented:
What was the code analysis too used?

nopara73 commented on WalletWasabi/Models/ChaumianCoinJoin/CcjRound.cs:
```diff
@@ -764,7 +747,7 @@ public int RemoveAlicesBy(AliceState state)
 					foreach (OutPoint input in alice.Inputs.Select(y => y.OutPoint))
 					{
-						GetTxOutResponse getTxOutResponse = await RpcClient.GetTxOutAsync(input.Hash, (int)input.N, includeMempool: true);
+						GetTxOutResponse getTxOutResponse = await RpcClient.GetTxOutAsync(input.Hash, (int)input.N);
```
Again, why?

nopara73 commented on WalletWasabi/TorSocks5/TorSocks5Client.cs:
```diff
@@ -40,14 +40,7 @@ public bool IsConnected
-					{
-						return true;
-					}
+					return TcpClient != null && TcpClient.Connected;
```
Why?

lontivero replied:
Because GetStream never returns a null, it is a redundant check.

nopara73 commented on WalletWasabi/Logging/Logger.cs:
```diff
@@ -24,7 +24,7 @@ public static class Logger
 		public static string FileEntryEncryptionPassword { get; private set; } = null;
 
-		private static long _logerFailed = Interlocked.Exchange(ref _logerFailed, 0);
+		private static long _logerFailed;
```
Could you initialize it with `=0`?

lontivero replied:
Sure. What's the point of doing that? I mean, the initial value is always, with no exception, zero.

nopara73 replied:
Readability.

nopara73 commented on WalletWasabi/Services/MemPoolBehavior.cs:
```diff
@@ -89,7 +89,7 @@ protected override void DetachCore()
 		private void ProcessTxPayload(TxPayload transactionPayload)
 		{
-			var transaction = transactionPayload.Object as Transaction;
+			var transaction = transactionPayload.Object;
```
Could you make it `Transaction transaction` instead of `var transaction`? For readability.

lontivero replied:
Sure.

nopara73 commented on WalletWasabi/Bases/ByteArraySerializableBase.cs:
```diff
@@ -23,7 +23,7 @@ protected ByteArraySerializableBase()
 		public string ToHex() => ToHex(false);
 
-		public string ToHex(bool xhhSyntax = false)
+		public string ToHex(bool xhhSyntax)
```
Could you reverse this change? This xhhSyntax is not something someone should know about.

nopara73 replied:
Oh, I see. Sorry, leave your modifications.

nopara73 replied:
Or maybe just delete the `ToHex()` function and leave only the `ToHex(bool xhhSyntax = false)` in it.

nopara73 commented on WalletWasabi/Bases/OctetSerializableBase.cs:
```diff
@@ -28,7 +28,7 @@ protected OctetSerializableBase()
 		public string ToHex() => ToHex(false);
 
-		public string ToHex(bool xhhSyntax = false)
+		public string ToHex(bool xhhSyntax)
```
Could you reverse this change? This xhhSyntax is not something someone should know about.

nopara73 commented:
Could you rollbackthe `Don't pass arguments matching default parameters` commit? It's there for readability.

nopara73 commented:
Could you roll back the `@lontivero
Remove redundant initializations` commit? In case of default initialization it's there for readability, in case of unused constructors it's there, because it can be used, it's just simply not used.

nopara73 commented:
Could you remove the switch conversions? Switch statements are evil. https://bizmonger.wordpress.com/2015/06/14/switch-statements-are-evil/

nopara73 commented:
And the #region stuff from `Fix seven finding CodeFactor` should be reversed too, after the constructors are put back into their place.

lontivero commented on WalletWasabi.Backend/Controllers/ChaumianCoinJoinController.cs:
```diff
@@ -153,13 +153,10 @@ public IActionResult GetStates()
 						}
 
-						GetTxOutResponse getTxOutResponse = await RpcClient.GetTxOutAsync(inputProof.Input.Hash, (int)inputProof.Input.N, includeMempool: true);
+						GetTxOutResponse getTxOutResponse = await RpcClient.GetTxOutAsync(inputProof.Input.Hash, (int)inputProof.Input.N);
```
Because `true` is the default value of that parameter so we don't need to pass it.

lontivero commented:
> What was the code analysis too used?

I did these changes using the **Jetbrains Rider built-in Resharper** code analysis tool. However, dotnet already has a very good tool for doing the same and I was playing for a while with it. It is easy to configure and make it run:

Edit the WalletWasabi.csproj file and add the following lines:

```xml
  <PropertyGroup>
    <CodeAnalysisRuleSet>../WalletWasabi.ruleset</CodeAnalysisRuleSet>
  </PropertyGroup>
```

Into the existing `ItemGroup` element containing the `PackageReference`s just add the following two additional references:
```xml
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.0" />
    <PackageReference Include="StyleCop.Analyzers" Version="1.0.2" />
```

And that's all. Now just type:

```bash
$ dotnet restore
$ dotnet build
```

My current `WalletWasabi.ruleset`  file looks like:
```xml
<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="Custom Rulset" Description="Custom Rulset" ToolsVersion="14.0">
    <Rules AnalyzerId="AsyncUsageAnalyzers" RuleNamespace="AsyncUsageAnalyzers">
        <Rule Id="UseConfigureAwait" Action="Warning" />
        <Rule Id="SA1027" Action="None" />
    </Rules>
    <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
        <Rule Id="SA1027" Action="None" />
    </Rules>
    
    <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
        <Rule Id="SA1305" Action="None" />
        <Rule Id="SA1412" Action="None" />
        <Rule Id="SA1600" Action="None" />
        <Rule Id="SA1609" Action="None" />
        <Rule Id="SA1101" Action="None" />
        <Rule Id="SA1027" Action="None" />
        <Rule Id="SA1200" Action="None" />
        <Rule Id="SA1208" Action="None" />
    </Rules>
</RuleSet>
```

nopara73 commented:
@lontivero

1. It's WIP, right? Since you didn't address many of my comments.  
2. Wouldn't it be better to make every change in a separate PR? It's already getting confusing.

lontivero commented:
>It's WIP, right? Since you didn't address many of my comments.

I've addressed most of your comments immediately by rebasing and dropping those commits that you didn't agree on. That's why there are no new commits like 'Revert commit xxxxxx'. OTOH, i was waiting for you comments to me comments.

> Wouldn't it be better to make every change in a separate PR? It's already getting confusing.

No problem, i will do that.

nopara73 commented:
Thanks.
